python/ray/_raylet.pyx

```diff
@@ -503,6 +520,7 @@ cdef execute_task(
         const c_vector[shared_ptr[CRayObject]] &c_args,
         const c_vector[CObjectID] &c_arg_reference_ids,
         const c_vector[CObjectID] &c_return_ids,
+        c_bool is_direct_call,
```

Comment:
The only thing this flag controls is how we store the return objects. Instead of introducing this can we define a new `TaskReturn` class that can either be a `ByReferenceReturn` that holds an object ID that the return should be put under or a `ByValueReturn` that holds a buffer that the return should be written into? This would make things more clear and also get rid of the `c_arg_reference_ids` entirely.

Reply:
I renamed this to `return_outputs_directly`, which should solve the clarity issue without needing to introduce a new class.